autograder/cli/users/upsert-file.py

```diff
@@ -0,0 +1,114 @@
+import sys
+
+import autograder.api.constants
+import autograder.api.users.upsert
+import autograder.cli.common
+import autograder.util.hash
+
+def run(arguments):
+    arguments = vars(arguments)
+
+    arguments['raw-users'] = _load_users(arguments['path'])
+    arguments['send-emails'] = not arguments['skip-emails']
+
+    result = autograder.api.users.upsert.send(arguments, exit_on_error = True)
+
+    autograder.cli.common.list_user_op_responses(result['results'], table = arguments['table'])
+    return 0
+
+def _load_users(path):
+    users = []
+
+    with open(path, 'r') as file:
+        lineno = 0
+        for line in file:
+            lineno += 1
+
+            line = line.strip()
+            if (line == ""):
+                continue
+
+            parts = line.split("\t")
```

Comment:
Go ahead and strip each field (I usually just do it once in a list comprehension).

The only not safe field we are doing this on is the password.
But, we can warn them in the help that all fields in the file will have leading and trailing whitespace removed.